src/webgpu/api/validation/buffer/create.spec.ts

```diff
@@ -26,21 +28,56 @@ g.test('size')
         kBufferSizeAlignment * 2,
       ])
   )
-  .unimplemented();
+  .fn(t => {
+    const { mappedAtCreation, size } = t.params;
+    const isValid = !mappedAtCreation || size % kBufferSizeAlignment === 0;
+    const usage = BufferUsage.COPY_SRC;
+    t.expectGPUError(
+      'validation',
+      () => t.device.createBuffer({ size, usage, mappedAtCreation }),
+      !isValid
+    );
+  });
+
+const BufferUsage = GPUConst.BufferUsage;
+const BufferUsageWithEmptyAndInvalid = { ...BufferUsage, NO_USAGE: 0, INVAILD: 0x8000 } as const;
+const listBufferUsage = Object.keys(BufferUsage) as [keyof typeof BufferUsage];
+const listBufferUsageWithEmptyAndInvalid = Object.keys(BufferUsageWithEmptyAndInvalid) as [
+  keyof typeof BufferUsageWithEmptyAndInvalid
+];
+const allowedBufferUsageSet = listBufferUsage.reduce(
+  (previousSet, currentUsage) => previousSet | BufferUsage[currentUsage],
+  0
+);
 
 g.test('usage')
-  .desc('Test combinations of (one to two?) usage flags.')
+  .desc('Test combinations of up to two usage flags are validated to be valid.')
```

Comment:
nit: specify "zero to two" so the reader can clearly see that 0 is tested.

Reply:
Done.